### PR TITLE
fix: 9559 issue android json parse

### LINKED ISSUE #9559: [Sentry] SyntaxError: JSON Parse error: Unexpected token: o
Sentry Issue: [METAMASK-MOBILE-2CVF](https://metamask.sentry.io/issues/5295138811/?referrer=github_integration)

```
SyntaxError: JSON Parse error: Unexpected token: o
  at emit (app:///index.android.bundle:22:16)
  at emit (app:///index.android.bundle:22:16)
...
(8 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

Fix issue on Android while handling notifications (JSON parse)

## **Related issues**

Fixes: [9559](https://github.com/MetaMask/metamask-mobile/issues/9559)

## **Manual testing steps**

1. Enable notifications feature (feature flag)
2. Submit any transaction within the app
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/util/notifications/types/notification/index.ts [hunk 1]
```diff
@@ -111,3 +111,11 @@ export interface MarketingNotificationData {
   route?: string;
   routeProps?: string;
 }
+
+export interface SimpleNotification {
+  title?: string | undefined;
```
Comment (Cal-L) [inline app/util/notifications/types/notification/index.ts:116]:
```suggestion
  title?: string;
```

File: app/util/notifications/types/notification/index.ts [hunk 2]
```diff
@@ -111,3 +111,11 @@ export interface MarketingNotificationData {
...
+  body?: string | undefined;
```
Comment (Cal-L) [inline app/util/notifications/types/notification/index.ts:117]:
```suggestion
  body?: string;
```

**Comment by Cal-L** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=metamask-mobile&pullRequest=9568) **Quality Gate failed**
> Failed conditions  [23.3% Coverage on New Code](https://sonarcloud.io/component_measures?id=metamask-mobile&pullRequest=9568&metric=new_coverage&view=list) (required ≥ 40%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=metamask-mobile&pullRequest=9568)

Let's increase test coverage for the notifications hook. That should be able to clear the threshold.

File: app/util/notifications/hooks/index.ts [hunk 3]
```diff
@@ -1,53 +1,67 @@
...
-       * Creates a channel (required for Android)
-       */
-      notifee.createChannel({
-        id: STORAGE_IDS.ANDROID_DEFAULT_CHANNEL_ID,
-        name: 'Default',
-        importance: AndroidImportance.HIGH,
-      });
+      notifee.onForegroundEvent(handleNotificationPressed);
+
+      setupAndroidChannel();
```
Comment (legobeat) [inline app/util/notifications/hooks/index.ts:58]: It seems like while this is Android-specific, it is being called on all platforms/targets?

File: app/util/notifications/hooks/index.ts [hunk 4]
```diff
@@ -1,53 +1,67 @@
...
+  EventType,
+} from '@notifee/react-native';
 import NotificationManager from '../../../core/NotificationManager';
 import Routes from '../../../constants/navigation/Routes';
+import { setupAndroidChannel } from '../setupAndroidChannels';
+import { SimpleNotification } from '../types';
 
 const useNotificationHandler = (
-  bootstrapInitialNotification: () => Promise<void>,
+  bootstrapAndroidInitialNotification: () => Promise<void>,
```
Comment (legobeat) [inline app/util/notifications/hooks/index.ts:12]: Is there a reason why the new name is preferred? The hook as such does not appear Android-specific, still.
```suggestion
  bootstrapInitialNotification: () => Promise<void>,
```

Comment (Jonathansoufer) [reply app/util/notifications/hooks/index.ts:12]: fixed.

File: app/util/notifications/hooks/index.ts (hunk 3 above)
Comment (Jonathansoufer) [reply app/util/notifications/hooks/index.ts:58]: added conditional.

File: .yarnrc [hunk 5]
```diff
@@ -1 +1,6 @@
-ignore-scripts true
\ No newline at end of file
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+ignore-scripts true
+yarn-path ".yarn/releases/yarn-1.22.22.cjs"
```
Comment (legobeat) [inline .yarnrc:6]:
```suggestion
ignore-scripts true
```
Unrelated change breaks `yarn setup` and we shouldn't commit the yarn binary in any case I think.

Comment (sethkfman) [reply .yarnrc]: Accidental push has been reverted

**Comment by NicolasMassart** [comment]
hi @Jonathansoufer, could you fix this PR description and an explicit unit test to prove that MetaMask/metamask-mobile#9559 is fixed by this PR please?
I would expect a unit test that uses a mock JSON similar to the one that made the app fail before to make sure we can consider this PR as a fix for the parsing issue. Thanks!

**Review by sethkfman — APPROVED** [review]
LGTM

**Pull request merged**